nltk/text.py

```diff
@@ -530,16 +534,50 @@ def dispersion_plot(self, words):
         from nltk.draw import dispersion_plot
 
         dispersion_plot(self, words)
-
-    def generate(self, words):
-        """
-        Issues a reminder to users following the book online
-        """
-        import warnings
-
-        warnings.warn(
-            'The generate() method is no longer available.', DeprecationWarning
-        )
+        
+    def build_ngram_lm(self, tokenized_sents, n=3):
+        train_data, padded_sents = padded_everygram_pipeline(n, tokenized_sents)
+        model = MLE(order=n)
+        model.fit(train_data, padded_sents)
+        return model
+        
+    def generate(self, length=100, text_seed=None, random_seed=42):
+        """
+        Print random text, generated using a trigram language model.
+        See also `help(nltk.lm)`.
+        
+        :param length: The length of text to generate (default=100)
+        :type length: int
+        
+        :param text_seed: Generation can be conditioned on preceding context.
+        :type text_seed: list(str)
+        
+        :param random_seed: A random seed or an instance of `random.Random`. If provided,
+        makes the random sampling part of generation reproducible.
+        :type random_seed: int
+        
+        """
+        # Create the model when using it the first time.
+        self.tokenized_sents = [sent.split(' ') for sent in 
+                                sent_tokenize(' '.join(self.tokens))]
+        if not hasattr(self, 'trigram_model'):
+            print("Building ngram index...", file=sys.stderr)
```

Comment:
Should we make this a logging call and let the caller decide where and how it gets reported? Or are folks coming into this with expectations to see printed output?

Reply:
In the old code, it prints to `stdout`, 73f7e7b#diff-f5cb9102f911e93ca5fefa00b56a1e3f. I'm not sure what to do with it and getting a logger here might be overkill =)

Reply:
ok, fair enough.